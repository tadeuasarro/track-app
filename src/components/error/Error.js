import PropTypes from 'prop-types';
import './error.css';

const Error = ({ error }) => {
  if (error.length === 0) return '';
  console.log(error);
  return (
    <div className="error-container">
      a
    </div>
  );
}

Error.propTypes = {
  error: PropTypes.array,
}

Error.defaultProps = {
  error: [],
}

export default Error;
