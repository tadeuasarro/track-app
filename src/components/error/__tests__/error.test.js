import { render, screen, act } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Error from '../Error';

describe('Rendering component', () => {
  it('creates an App component', () => {
    act(() => {
      render(
        <Error error={['this is some error']} />,
      );
    });
    // eslint-disable-next-line
    const element = screen.getByText('This is some error');
  });
});

describe('Display', () => {
  it('renders correctly', () => {
    const comp = renderer.create(
      <Error error={['this is some error']} />,
    ).toJSON();
    expect(comp).toMatchSnapshot();
  });
});
