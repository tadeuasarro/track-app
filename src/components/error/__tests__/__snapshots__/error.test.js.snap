// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<div
  className="error-container"
>
  <p>
    This is some error
  </p>
</div>
`;
